### DISCUSSION TITLE
Outbox pattern maybe lock in OutboxMessages table

### DISCUSSION THREAD

**Discussion by SenSyIot** [kgrzybek/sample-dotnet-core-cqrs-api#32] [Q&A] [answered]
If the requests are too many. May cause a lock in the OutboxMessages table and causes slowness in other tables.
What could be your solution?
Do I need a message table for each AggrigateRoot?

**Comment by kgrzybek** [comment, marked as answer]
If you experience performance issues, you can use two tables approach:
1. One for messages
2. One for messages consumers

Then you can use the last processed event number and partition key to increase performance and allow parallel processing. More info you can find here: https://event-driven.io/en/outbox_inbox_patterns_and_delivery_guarantees_explained/
